From 8ec358c67ae331ff5be6618439d2532b9f14883c Mon Sep 17 00:00:00 2001
From: Suresh Gupta <suresh.gupta@xilinx.com>
Date: Fri, 5 Oct 2018 19:42:56 +0530
Subject: [PATCH 0449/1851] v4l: xilinx-multi-scaler: Fix buffer type in
 enum_fmt_vid_out

commit 0aa69ce5378ef82b2399ba58502a68edc1eedf7a from
https://github.com/Xilinx/linux-xlnx.git

Driver is checking with wrong buffer type in enum_fmt_vid_out()
implementation.
This patch corrects the buffer type to be checked against.

Signed-off-by: Suresh Gupta <suresh.gupta@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Reviewed-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-multi-scaler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-multi-scaler.c b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
index 66c322eba0fb..98cda9a94240 100644
--- a/drivers/media/platform/xilinx/xilinx-multi-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-multi-scaler.c
@@ -1264,7 +1264,7 @@ static int xm2msc_enum_fmt_vid_out(struct file *file, void *fh,
 {
 	struct xm2msc_chan_ctx *chan_ctx = fh_to_chanctx(fh);
 
-	if (f->type != V4L2_BUF_TYPE_VIDEO_CAPTURE_MPLANE)
+	if (f->type != V4L2_BUF_TYPE_VIDEO_OUTPUT_MPLANE)
 		return -EINVAL;
 
 	return enum_fmt(chan_ctx->xm2msc_dev, f);
-- 
2.31.1

